### PR TITLE
feat(core): Add configurable request timeout capability.

### PR DESCRIPTION
**Pull request by OmarAlJarrah**
### :pencil: Description
- Add configurable request timeout capability.

### :link: Related Issues
-

### CODE REVIEW

File: src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/provider/OpenWorldConfigurationProvider.kt [hunk 1]
```diff
@@ -18,16 +18,20 @@ package com.expediagroup.openworld.sdk.core.configuration.provider
...
  * @property name The name of the provider.
  * @property endpoint The API endpoint to use for requests.
  * @property authEndpoint The API endpoint to use for authentication.
+ * @property requestTimeoutMillis The API response timeout to use for requests.
  */
 internal object OpenWorldConfigurationProvider : ConfigurationProvider {
     override val name: String = "OpenWorld Configuration Provider"
     override val endpoint: String = "https://api.expediagroup.com/"
     override val authEndpoint: String = "${endpoint}identity/oauth2/v2/token/"
+    override val requestTimeoutMillis: Long = Constant.TEN_SECONDS_IN_MILLIS
```
Comment (mohnoor94) [inline src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/provider/OpenWorldConfigurationProvider.kt]: As the default value for both providers (`OpenWorldConfigurationProvider` and `OpenWorldConfigurationProvider`) is the same, let's move it to the `ConfigurationProvider` and remove the field from these providers.

File: src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/ClientConfiguration.kt [hunk 2]
```diff
@@ -21,11 +21,13 @@ interface ClientConfiguration {
...
     val endpoint: String?
+    val requestTimeoutMillis: Long?
 
     /** Build a [RuntimeConfigurationProvider] from a [ClientConfiguration]. */
     fun toProvider(): RuntimeConfigurationProvider = RuntimeConfigurationProvider(
         key = key,
         secret = secret,
-        endpoint = endpoint
+        endpoint = endpoint,
+        requestTimeoutMillis = requestTimeoutMillis
```
Comment (mohnoor94) [inline src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/ClientConfiguration.kt]: Let's rename it to `requestTimeout` over all the code, and name the parameter as `milliseconds` in the builders.

File: src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/OpenWorldClientConfiguration.kt [hunk 3]
```diff
@@ -30,7 +30,8 @@ data class OpenWorldClientConfiguration(
     override val key: String? = null,
     override val secret: String? = null,
     override val endpoint: String? = null,
-    val authEndpoint: String? = null
+    val authEndpoint: String? = null,
```
Comment (mohnoor94) [inline src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/OpenWorldClientConfiguration.kt]: Let's keep any added config to the last (switch the order between `authEndpoint` and `requestTimeoutMillis`).

**Review by mohnoor94 — CHANGES_REQUESTED** [review]
**Important**: Please update the `Builder` class (inside `Client`) with the new configuration.

File: src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/ClientConfiguration.kt (hunk 2 above)
Comment (OmarAlJarrah) [reply src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/ClientConfiguration.kt]: Is not `requestTimeoutMillis` more intuitive for any developer to figure out that the value is measured with milliseconds when used all over the code? 

Especially that the [same name is used](https://api.ktor.io/ktor-client/ktor-client-core/io.ktor.client.plugins/-http-timeout/-http-timeout-capability-configuration/request-timeout-millis.html) in ktor for the value.

Comment (mohnoor94) [reply src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/ClientConfiguration.kt]: Yes, but the only way to provide this by the partner will be via our builder, which has the parameter name as `milliseconds` as I added in another comment. Both of them will tell the same info in a more elegant way (in addition, it's the common way to provide this value in milliseconds).

_Anyway, if you feel the other way around, go for it, it's not a big deal._

File: src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/OpenWorldClientConfiguration.kt (hunk 3 above)
Comment (OmarAlJarrah) [reply src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/OpenWorldClientConfiguration.kt]: Done

File: src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/provider/OpenWorldConfigurationProvider.kt (hunk 1 above)
Comment (OmarAlJarrah) [reply src/main/kotlin/com/expediagroup/openworld/sdk/core/configuration/provider/OpenWorldConfigurationProvider.kt]: Done

File: src/main/kotlin/com/expediagroup/openworld/sdk/core/plugin/httptimeout/HttpTimeoutConfiguration.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ */
+package com.expediagroup.openworld.sdk.core.plugin.httptimeout
+
+import com.expediagroup.openworld.sdk.core.plugin.KtorPluginConfiguration
+import io.ktor.client.HttpClientConfig
+import io.ktor.client.engine.HttpClientEngineConfig
+
+internal data class HttpTimeoutConfiguration(
+    override val httpClientConfiguration: HttpClientConfig<out HttpClientEngineConfig>,
+    val milliseconds: Long
```
Comment (mohnoor94) [inline src/main/kotlin/com/expediagroup/openworld/sdk/core/plugin/httptimeout/HttpTimeoutConfiguration.kt]: Let's use `requestTimeout` here (and on the static factory method below).

**Review by mohnoor94 — CHANGES_REQUESTED** [review]

**Review by mohnoor94 — APPROVED** [review]
Nice work, thank you!

**Pull request merged**